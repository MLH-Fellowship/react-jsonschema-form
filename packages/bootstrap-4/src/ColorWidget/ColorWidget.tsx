import React from "react";
import TextWidget, { TextWidgetProps } from "../TextWidget";

export interface ColorWidget extends TextWidgetProps {
    type?: string;
}

const ColorWidget = (props: TextWidgetProps) => {
  return <TextWidget type="color" {...props} />;
};

export default ColorWidget;
