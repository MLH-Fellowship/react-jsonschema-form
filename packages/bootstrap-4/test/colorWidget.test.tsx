import React from "react";
import ColorWidget from "../src/ColorWidget";
import { JSONSchema7 } from "json-schema";
import renderer from "react-test-renderer";

describe("ColorWidget", () => {
  test("simple", () => {
    const schema: JSONSchema7 = {
      type: "array",
      items: {
        type: "string",
      },
    };
    const mockHandler = (): void => void 0;
    const tree = renderer
      .create(
        <ColorWidget
          schema={schema}
          required
          disabled={false}
          readonly
          autofocus
          label="Some simple label"
          onChange={mockHandler}
          onBlur={mockHandler}
          onFocus={mockHandler}
          multiple={false}
          rawErrors={[""]}
          value="value"
          options={{}}
          formContext={{}}
          id="_id"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
