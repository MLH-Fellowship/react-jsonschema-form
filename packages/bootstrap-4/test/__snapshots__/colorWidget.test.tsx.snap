// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorWidget simple 1`] = `
<div
  className="mb-0 form-group"
>
  <label
    className="form-label"
    htmlFor="_id"
  >
    Some simple label
    *
  </label>
  <input
    autoFocus={true}
    className="form-control"
    disabled={true}
    id="_id"
    multiple={false}
    name="nodejs"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    type="color"
    uiSchema={Object {}}
    value="value"
  />
</div>
`;
